File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Dispatchs/Commands/TaskPerformerCommand.js
SIE.defineCommand('SIE.Web.MES.TaskManagement.Dispatchs.Commands.TaskPerformerCommand', {
    meta: { text: "可执行对象保存", group: "edit" },
    execute: function (view, source) {
        var me = view;
        var dispatchTaskId = source.data.DispatchTaskId;
        var selectedIds = view.getSelectionIds();
        me.execute({
            command: Ext.getClassName(this),
            data: source.data,
            success: function (res) {
                var errMsg = res.Result;
                if (errMsg !== '保存成功')
                    SIE.Msg.showMessage(errMsg);
               // else
              //      view.reloadData();
                if (source.data.Status == 0)
                    return;
                var adoTypeBoxControl = Ext.getCmp("adoTypeBoxId");
                var adoNameBoxControl = Ext.getCmp("adoNameBoxId");
                var adoType = adoTypeBoxControl.value;
                var adoName = adoNameBoxControl.value;
                if (adoName == null)
                    adoName = "";
                SIE.invokeDataQuery({
                    type: "SIE.Web.MES.TaskManagement.Dispatchs.DispatchDataQueryer",
                    method: "GetTaskPerformerInfo",
                    params: [selectedIds, dispatchTaskId, adoType, adoName],
                    async: false,
                    token: view.token,
                    callback: function (res) {
                        if (res.Success) {
                            var taskPerformerInfo = res.Result;
                            view.isSelectedTaskPerformer = taskPerformerInfo.IsSelectedTaskPerformer;
                            var dragControl = Ext.getCmp('taskPerfomerId');
                            var grid1Id = dragControl.items.items[0].id;
                            var grid1Control = Ext.getCmp(grid1Id);
                            var store = grid1Control.getStore();
                            store.setData(taskPerformerInfo.AdoInfos);
                            grid1Control.setStore(store);
                        }
                    }
                });
            }
        });
    }
});

File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Dispatchs/Commands/TaskPerformerCommand.cs
using SIE.MES.TaskManagement.Dispatchs;
using SIE.Security;
using SIE.Web.Command;

namespace SIE.Web.MES.TaskManagement.Dispatchs.Commands
{
    /// <summary>
    /// 保存派工任务可执行对象
    /// </summary>
    [JsCommand("SIE.Web.MES.TaskManagement.Dispatchs.Commands.TaskPerformerCommand")]
    [AllowAnonymous]
    public class TaskPerformerCommand : SaveCommand
    {
        /// <summary>
        /// 派工任务可执行对象命令执行方法
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>保存结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            string errMsg = string.Empty;
            var adoInfo = args.Data.ToJsonObject<AdoInfo>();
            errMsg = RT.Service.Resolve<DispatchController>().SaveTaskPerformer(adoInfo);
            if (errMsg.Length == 0)
                return "保存成功";
            else
                return errMsg;
        }
    }
}
